dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -368,6 +374,100 @@ public static Doi getDoiFromDatabase(DoiType doiType, DoiInitiator doiInitiator,
         return doi;
     }
 
+    /**
+     * Searches for Zenodo DOIs issued against the GitHub repository <code>gitHubRepo</code>.
+     * <p>
+     * The Zenodo API returns an ElasticSearch response. What we do is:
+     *
+     * <ol>
+     *     <li>Make a request to <code>https://zenodo.org/api/records?q=[urlencoded gitHubRepo]</code></li>
+     *     <li>Look in the response for hits.hits[].metadata.relatedIdentifiers[] for an identifier that is <code>https://github.com/[org]/[repo]/tree/[tag]</code></li>
```

Comment:
Did some digging and I think the second step can be removed if you filter the identifier in the first request to Zenodo. The query uses [Elasticsearch's query string syntax](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-string-syntax) which allows you to specify fields to search in and you can use regex.

For example, I enabled the Zenodo/GitHub integration for repository https://github.com/kathy-t/ghapps-single-workflow/tree/githubzenodo and it created a DOI for the `githubzenodo` release.

I can search for it using the following query which searches in the related identifiers and returns 1 result vs the 35 results that the original query would've returned:

https://sandbox.zenodo.org/api/records?q=metadata.related_identifiers.identifier:%2Fhttps%3A%5C%2F%5C%2Fgithub%5C.com%5C%2Fkathy-t%5C%2Fghapps-single-workflow%5C%2Ftree%5C%2F.%2B%2F


The above query is URL encoded, the unencoded URL is the following where the regex is between the slashes (note that the unencoded URL will fail):

https://sandbox.zenodo.org/api/records?q=metadata.related_identifiers.identifier:/https:\/\/github\.com\/kathy-t\/ghapps-single-workflow\/tree\/.+/

Reply:
Nice find!